src/main/java/ua/goit/gojava32/kickstarter/controller/UserController.java

```diff
@@ -0,0 +1,96 @@
+package ua.goit.gojava32.kickstarter.controller;
+
+import org.apache.commons.codec.digest.DigestUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.servlet.ModelAndView;
+import ua.goit.gojava32.kickstarter.model.User;
+import ua.goit.gojava32.kickstarter.service.ProjectService;
+import ua.goit.gojava32.kickstarter.service.SendMail;
+import ua.goit.gojava32.kickstarter.service.UserService;
+
+import javax.servlet.http.HttpServletRequest;
+import java.security.Principal;
+
+@Controller
+public class UserController {
+
+  @Autowired
+  UserService userService;
+
+  @Autowired
+  ProjectService projectService;
+
+  @RequestMapping(value = {"/login","/login_page"}, method = RequestMethod.GET)
+  @ResponseBody
+  public ModelAndView login(){
+    return new ModelAndView("login_page");
+  }
+
+  @RequestMapping(value = "/registration", method = RequestMethod.POST)
+  @ResponseBody
+  public ModelAndView registration(HttpServletRequest request,
+                               @RequestParam("name") String name,
+                               @RequestParam("email") String email,
+                               @RequestParam("password") String password) {
+
+    String md5password = DigestUtils.md5Hex(password);
+
+    User user = userService.findUserByEmail(email);
+    ModelAndView vm;
+
+    if (user == null) {
+      user = new User();
+      user.setName(name);
+      user.setEmail(email);
+      user.setPassword(md5password);
+      userService.add(user);
+      String domain = request.getRequestURL().toString();
+      domain = domain.substring(0, domain.length() - 13);///registration
```

Comment:
I don't think `-13` works for everyone(or if it does refactor it as a constant), also good candidate to be refactored out into a function